sdks/python/apache_beam/io/gcp/bigquery.py

```diff
@@ -1841,15 +2118,29 @@ class ReadFromBigQuery(PTransform):
     on GCS, and then reads from each produced file. File format is Avro by
     default.
 
+  .. warning::
+      DATETIME columns are parsed as strings in the fastavro library. As a
+      result, such columns will be converted to Python strings instead of native
+      Python DATETIME types.
+
   Args:
+    method: The method to use to read from BigQuery. It may be EXPORT or
+      DIRECT_READ. EXPORT invokes a BigQuery export request
+      (https://cloud.google.com/bigquery/docs/exporting-data). DIRECT_READ reads
+      directly from BigQuery storage using the BigQuery Read API
+      (https://cloud.google.com/bigquery/docs/reference/storage). If
+      unspecified, the default is currently EXPORT.
+    use_fastavro (bool): If method is `DIRECT_READ` and :data:`True`, the
```

Comment:
Can we rename this to "use_fastavro_for_direct_read" for clarity ?

Also, why do you need to support both Avro and Fastavro libraries ?

Reply:
Renamed the flag.

I intend to remove the non-fastavro parsing in a follow-up PR where the Storage API source and the core avroio source are combined together. And also add integration tests in the follow-up.

Reply:
I don't know that we'll be able to combine the Avro and storage API stream sources -- I was expecting that we could remove support for the non-fastavro path from both at the same time, though.

Reply:
I think fastavro is the default for Avro source (Python3) for better perf and other issues in the Avro library. @tvalentyn might have more context.